src/managers/resources/emg.rs

```diff
@@ -8,18 +11,67 @@ use crate::managers::macros::parse_channel_data;
 use crate::request::TaskData::EmgData;
 use crate::sgcp;
 use crate::sgcp::emg::*;
-use crate::todo;
+use crate::managers::resources::adc;
+use crate::config::Config;
 use anyhow::Error;
 use anyhow::Result;
 use anyhow::anyhow;
 use log::*;
 
-/// Represents an EMG resource
-pub struct Emg {}
+
+use rppal::gpio::{Gpio, OutputPin};
+use rppal::spi::{Bus, Mode, SlaveSelect, Spi};
+
+pub struct Emg {
+    spi: Spi,
+    buffer_size: usize,
+    inner_threshold: u16,
+    outer_threshold: u16,
+    cs_pin: OutputPin,
+    inter_channel_sample_duration: u64, // different from sampling speed, this is the time between reading the inner and outer channels
+}
 
 impl Resource for Emg {
     fn init() -> Self {
-        Emg {} // stub
+
+        let emg_config = Config::global()
+        .emg_sensor
+        .as_ref()
+        .expect("Expected emg config to be defined");
+
+        let spi = Spi::new(Bus::Spi0, SlaveSelect::Ss0, 500_000, Mode::Mode0);
+        if spi.is_err() {
+            error!("Failed to initialize SPI: {:?}", spi.err());
+            panic!("Failed to initialize SPI");
+        }
```

Comment:
```suggestion
        let spi = Spi::new(Bus::Spi0, SlaveSelect::Ss0, 500_000, Mode::Mode0)
        .expect("Failed to initialize SPI");
```
This would include the error details.